Src/EB/AMReX_EB2_2D_C.cpp

```diff
@@ -252,19 +272,55 @@ void build_cells (Box const& bx, Array4<EBCellFlag> const& cell,
 
             // remove small cells
             if (vfrac(i,j,0) < small_volfrac) {
-                vfrac(i,j,0) = 0.0;
-                vcent(i,j,0,0) = 0.0;
-                vcent(i,j,0,1) = 0.0;
-                barea(i,j,0) = 0.0;
-                bcent(i,j,0,0) = -1.0;
-                bcent(i,j,0,1) = -1.0;
-                bnorm(i,j,0,0) = 0.0;
-                bnorm(i,j,0,1) = 0.0;
-                cell(i,j,0).setCovered();
+               set_covered(i,j,cell,vfrac,vcent,barea,bcent,bnorm);
             }
         }
     });
 
+    // set cells in the extended region to covered if the
+    // corresponding cell on the domain face is covered
+    if(extend_domain_face) {
```

Comment:
We don't need to launch a kernel if the box is not at the domain boundary.  Moreever, `Geometry` cannot be used on device.  How about something like this here?
```
Box gdomain = geom.Domain();
for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
    if (geom.isPeriodic(idim)) {
        gdomain.setSmall(idim,std::min(gdomain.smallEnd(idim),bxg1.smallEnd(idim)));
        gdomain.setBig(idim,std::max(gdomain.bigEnd(idim),bxg1.bigEnd(idim)));
    }
}
if (not gdomain.contains(bxg1)) {
    AMREX_HOST_DEVICE_FOR_3D ( bxg1, i, j, k,
   {
       // No need to use geom
       // use gdomain
   });
}
```